api/app/Listeners/StoreApplicationSnapshot.php

```diff
@@ -35,10 +35,15 @@ public function handle(ApplicationSubmitted $event)
             'expectedClassifications',
             'expectedGenericJobTitles',
             'awardExperiences',
+            'awardExperiences.skills',
             'communityExperiences',
+            'communityExperiences.skills',
             'educationExperiences',
+            'educationExperiences.skills',
             'personalExperiences',
-            'workExperiences'
+            'personalExperiences.skills',
+            'workExperiences',
+            'workExperiences.skills'
```

Comment:
Yeah, so the skills are actually two relations away from user.    

user -> experience -> skill

If you just loaded user you would get no relations.  With awardExperiences you get that model but none of `its` relations so the awardExperiences.skills was still lazy loaded.  With awardExperiences.skills listed in the `with` that 2nd degree relation is also eagerly loaded.

Reply:
Yeah, I figured out how to run xdebug during this PR!  :partying_face: I left some notes in  #1359.

So, this code definitely fires a lazy load during the `$experience->skills()` but Eloquent doesn't seem to care.  Maybe because it's a MorphToMany?  I'm really not sure.  Adding it to the `has` in the `collectExperiences` function doesn't prevent the lazy load, either.  I'll have to spend a bit of time on this one.

Reply:
how about here, does it somehow load skills when in the resource collector process then?

![image](https://user-images.githubusercontent.com/40485260/213323608-54e3cf26-947a-4e64-8747-f7fe619ae70f.png)

Reply:
I'm not 100% sure why this load wasn't flagged by the checker but my best guess is that it didn't error because it was able to pluck the `id` from the experience_skill pivot table without actually loading the skill model from the skills table.  I've pushed up a refactor in [Fix lazy load in HasEssentials](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5381/commits/a76fb815a7a3c61c61d4388ecf8b9d6ec81b492a) that eagerly loads the skills though I'm not sure it's totally necessary.

Reply:
Looks a lot cleaner to me regardless if it was necessary 😜